node/pkg/altpub/alternate_pub.go

```diff
@@ -0,0 +1,559 @@
+func (ap *AlternatePublisher) createClient() (*http.Client, error) {
+	defTrans, ok := http.DefaultTransport.(*http.Transport)
+	if !ok {
+		return nil, errors.New("failed to cast default transport for alternate publisher")
+	}
+
+	t := defTrans.Clone()
+	t.MaxIdleConns = MaxIdleConns
+	t.MaxConnsPerHost = MaxConnsPerHost
+	t.MaxIdleConnsPerHost = MaxIdleConnsPerHost
+
+	return &http.Client{
+		Timeout:   HttpClientTimeout,
+		Transport: t,
+	}, nil
+}
+
+// startHttpWorkers starts all of the workers in the HTTP worker pool.
+func (ap *AlternatePublisher) startHttpWorkers(ctx context.Context, client *http.Client, errC chan error) {
+	numWorkers := NumWorkersPerEndpoint * len(ap.endpoints)
+	for count := range numWorkers {
+		worker := &HttpWorker{ap, client, count}
+		common.RunWithScissors(ctx, errC, "alt_pub_worker", worker.httpWorker)
+	}
+}
+
+// httpWorker is the entrypoint for an HTTP worker. Each httpWorker is responsible for posting an HTTP request and waiting for the response.
+func (w *HttpWorker) httpWorker(ctx context.Context) error {
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case req, ok := <-w.ap.httpWorkerChan:
+			if !ok {
+				return fmt.Errorf("httpWorker failed to read request because the channel has been closed")
+			}
+			if err := w.ap.httpPost(ctx, w.client, req); err != nil {
+				// These errors are not fatal, so just log it an continue.
+				w.ap.logger.Error("failed to post http request", zap.String("endpoint", req.ep.label), zap.Error(err))
+			}
+		}
+	}
+}
+
+// httpPost actually posts an HTTP request and waits for the response. It pegs metrics based on the results.
+func (ap *AlternatePublisher) httpPost(ctx context.Context, client *http.Client, req *HttpRequest) error {
+	channelDelay.WithLabelValues(req.ep.label).Observe(float64(time.Since(req.start).Microseconds()))
+
+	// Create the HTTP POST request using our context so it can be interrupted.
+	// Note that we are not using a timeout context because the HTTP client already has a timeout.
+	// Note that we make a copy of the payload because `bytes.NewBuffer` takes ownership of the data, and the same data might be in multiple requests.
+	r, err := http.NewRequestWithContext(ctx, "POST", req.url, bytes.NewBuffer(bytes.Clone(req.data)))
+	if err != nil {
+		requestFailed.WithLabelValues(req.ep.label, "create_failed").Inc()
+		return fmt.Errorf("create failed: %w", err)
+	}
+	r.Header.Add("Content-Type", "application/octet-stream")
+
+	start := time.Now()
+	resp, err := client.Do(r)
+	if err != nil {
+		if strings.Contains(err.Error(), "connection refused") {
+			requestFailed.WithLabelValues(req.ep.label, "connection_refused").Inc()
+		} else {
+			requestFailed.WithLabelValues(req.ep.label, "post_failed").Inc()
+		}
+		return fmt.Errorf("post failed: %w", err)
+	}
+	stop := time.Now()
+	resp.Body.Close()
+
+	// Peg the appropriate metric, based on the request result.
+	if resp.StatusCode < 300 {
```

Comment:
Isn't 200 -> 299 the actual range for success?

Reply:
The 100 range is info. Are you suggesting I treat those as errors?

Reply:
FWIW, this is in the go example https://pkg.go.dev/net/http#example-Get